### PR TITLE
feat(docs): add new view for design tokens

### PR DESCRIPTION
**Pull request by vepor**

### CODE REVIEW

File: docs/src/components/DesignTokens/components/DesignTokensTable.tsx [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+
+const DesignTokensTable = ({
+  tokens,
+  filter,
+  platform,
+  showDeprecated,
+  tableName,
+  showVariantColumn,
+}: {
+  tokens: Array<Token>;
```
Comment (mainframev) [inline docs/src/components/DesignTokens/components/DesignTokensTable.tsx:45]:
```suggestion
  tokens: Token[];
```

File: docs/src/components/DesignTokens/components/DesignTokensTable.tsx [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+  filter: string;
+  platform: keyof typeof Platforms;
+  showDeprecated?: boolean;
+  tableName: string;
+  showVariantColumn?: boolean;
+}) => {
+  const [variantFilterValue, setVariantFilterValue] = React.useState<Array<string>>([]);
```
Comment (mainframev) [inline docs/src/components/DesignTokens/components/DesignTokensTable.tsx:52]:
```suggestion
  const [variantFilterValue, setVariantFilterValue] = React.useState<string[]>([]);
```

File: docs/src/components/DesignTokens/components/DesignTokensTable.tsx [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+    .filter(Boolean)
+    .filter((v, i, a) => a.indexOf(v) === i)
+    .map(v => ({ key: v, name: camelCase(v) }));
+
+  const filteredTokens = tokens
+    .filter(({ name }) => name.toLowerCase().indexOf(filter) !== -1)
+    .filter(({ value }) => {
+      if (showDeprecated) {
+        return true;
+      }
```
Comment (mainframev) [inline docs/src/components/DesignTokens/components/DesignTokensTable.tsx:73]:
```suggestion
      if (showDeprecated) return true;
```

File: docs/src/components/Palette/index.tsx [hunk 4]
```diff
@@ -15,49 +16,54 @@ export const isLight = (hex: string) => {
...
-  }
-  return false;
+  return hsp > 140;
 };
 
-export type tokenString = keyof typeof defaultTokens;
-
 interface PaletteProps {
-  colors: tokenString[];
+  colors: Array<string>;
```
Comment (mainframev) [inline docs/src/components/Palette/index.tsx:23]:
```suggestion
  colors: string[];
```

File: docs/src/components/Palette/index.tsx [hunk 5]
```diff
@@ -15,49 +16,54 @@ export const isLight = (hex: string) => {
...
 }
 
-const normalizeName = (tokenName: string) =>
-  tokenName
-    .replace(/^palette/, "")
-    .replace(/^color/, "")
-    .replace(/([a-z])([A-Z])/, "$1 $2")
-    .replace("Hover", ":hover")
-    .replace("Active", ":active");
+const normalizeName = (name: Array<string>) => name.map(v => upperFirst(String(v))).join(" ");
```
Comment (mainframev) [inline docs/src/components/Palette/index.tsx:27]:
```suggestion
const normalizeName = (name: string[]) => name.map(v => upperFirst(String(v))).join(" ");
```

File: docs/src/utils/camelCase.ts [hunk 6]
```diff
@@ -0,0 +1,11 @@
+function camelCase(str: string): string {
```
Comment (mainframev) [inline docs/src/utils/camelCase.ts:1]: do not forget to replace that function with those from `lodash` ;)

File: docs/src/components/DesignTokens/GlobalDesignTokens.tsx [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+    );
+    return { [category]: categoryTokens };
+  }),
+);
+
+const GlobalDesignTokens = () => {
+  const [filter, setFilter] = React.useState<string>("");
+  const [showDeprecated, setShowDeprecated] = React.useState<boolean>(false);
+  const [platform, setPlatform] = React.useState<keyof typeof Platforms>("javascript");
+  const [selectedCategories, setSelectedCategories] = React.useState<Array<string>>([]);
```
Comment (mainframev) [inline docs/src/components/DesignTokens/GlobalDesignTokens.tsx:65]:
```suggestion
  const [selectedCategories, setSelectedCategories] = React.useState<string[]>([]);
```

File: docs/src/components/Palette/CopyButton.tsx [hunk 8]
```diff
@@ -1,6 +1,5 @@
-import React from "react";
+import * as React from "react";
```
Comment (mainframev) [inline docs/src/components/Palette/CopyButton.tsx:1]:
```suggestion
import React from "react";
```

File: docs/src/components/DesignTokens/GlobalDesignTokens.tsx [hunk 9]
```diff
@@ -0,0 +1,141 @@
...
+      </Box>
+      <OptionsFilter
+        label="Selected categories"
+        value={selectedCategories}
+        options={groupedCategories.map(c => ({ key: c, name: GlobalCategories[c] }))}
+        onChange={handleSelectCategory}
+      />
+      <Separator />
+      {Object.keys(globalTokens).map(category => {
+        const value = globalTokens[category];
```
Comment (mainframev) [inline docs/src/components/DesignTokens/GlobalDesignTokens.tsx:118]:
```suggestion
      {Object.entries(globalTokens).map(([category, value]) => {
```

File: docs/src/components/DesignTokens/components/DesignTokenColor.tsx [hunk 10]
```diff
@@ -0,0 +1,146 @@
+import * as React from "react";
```
Comment (mainframev) [inline docs/src/components/DesignTokens/components/DesignTokenColor.tsx:1]:
```suggestion
import React from "react";
```

File: docs/src/components/DesignTokens/components/DesignTokenIcon.tsx [hunk 11]
```diff
@@ -0,0 +1,65 @@
+import * as React from "react";
```
Comment (mainframev) [inline docs/src/components/DesignTokens/components/DesignTokenIcon.tsx:1]:
```suggestion
import React from "react";
```

File: docs/src/components/DesignTokens/components/DesignTokenValue.tsx [hunk 12]
```diff
@@ -0,0 +1,82 @@
+import * as React from "react";
```
Comment (mainframev) [inline docs/src/components/DesignTokens/components/DesignTokenValue.tsx:1]:
```suggestion
import React from "react";
```

File: docs/src/components/DesignTokens/components/DesignTokensTable.tsx [hunk 13]
```diff
@@ -0,0 +1,177 @@
+import * as React from "react";
```
Comment (mainframev) [inline docs/src/components/DesignTokens/components/DesignTokensTable.tsx:1]:
```suggestion
import React from "react";
```

File: docs/src/components/DesignTokens/components/DesignTokensTable.tsx [hunk 14]
```diff
@@ -0,0 +1,177 @@
...
+  };
+
+  const variants = tokens
+    .map(({ value }) => value?.schema?.variant)
+    .filter(Boolean)
+    .filter((v, i, a) => a.indexOf(v) === i)
+    .map(v => ({ key: v, name: camelCase(v) }));
+
+  const filteredTokens = tokens
+    .filter(({ name }) => name.toLowerCase().indexOf(filter) !== -1)
```
Comment (mainframev) [inline docs/src/components/DesignTokens/components/DesignTokensTable.tsx:69]:
```suggestion
    .filter(({ name }) => name.toLowerCase().includes(filter))
```

File: docs/src/components/Palette/index.tsx [hunk 15]
```diff
@@ -1,11 +1,12 @@
-import React from "react";
+import * as React from "react";
```
Comment (mainframev) [inline docs/src/components/Palette/index.tsx:1]:
```suggestion
import React from "react";
```

**Review by mainframev — COMMENTED** [review]
I walked through code, just small things, which can be easily fixed. Looked on the `design tokens` page, looks awesome. Here are few things which can be added later, so I will still write them here, so we will not forget and for others

- I think it would be nice to have a button to drop selected items, like if I selected 5 categories and would like to deselect them, instead of clicking on them again or adding them all ass selected.
- It would be better to show the copy button for token name and the value when I hovered on the whole line, not only on name or value. 
- Inputs on the mobile are not placed on the whole width each under each. Add prop to `Stack` will solve it :) 
- Categories buttons are should be wrapped into `Stack`.
- Update the Table to the latest design, Grid instead of Table maybe 🤷🏻

File: docs/src/components/DesignTokens/ComponentDesignTokens.tsx [hunk 16]
```diff
@@ -0,0 +1,96 @@
...
+            />
+          );
+        }
+        return null;
+      })}
+    </Stack>
+  );
+};
+
+export default GlobalDesignTokens;
```
Comment (vepor) [inline docs/src/components/DesignTokens/ComponentDesignTokens.tsx:96]:
```suggestion
export default ComponentDesignTokens;
```

**Review by mainframev — APPROVED** [review]

**Pull request merged**